### DISCUSSION TITLE
[BP #77 #81 #78 #83] Rassembler les pratiques « minifier » et « compresser »

### DISCUSSION THREAD

**Discussion by rlemaire** [cnumr/best-practices#21] [Bonnes pratiques] [answered]
Pour éviter les redondances on peut avoir « minifier les fichiers html, css et js » et « compresser les fichiers »

**Comment by arnolem** [comment]
C'est à discuter car il s'agit bien de notions différentes mais surtout, ça ne se configure pas forcément au même endroit.
Je peux minifier un fichier CSS avec un script local (ex : gulp) et compresser les fichiers (GZIP/DEFLATE) à la volé avec un configuration sur mon serveur web. Dans cet exemple, c'est même deux personnes différentes qui pourront être en charge de l'optimisaition.

**Reply by rlemaire** [reply]
Alors j'ai peut-être été trop concis mais l'idée est de ne pas avoir deux pratiques différentes pour minifier, par exemple « Minifier les fichiers CSS » et « Minifier les fichiers JavaScript » mais une seule « Minifier les fichiers CSS, JS et HTML ». 
Et dans le même esprit, ne pas avoir « Compresser les librairies CSS et Javascript » et « Compresser la sortie HTML » mais seulement « Compresser les fichiers CSS, JS et HTML ».
Du coup cela fait deux pratiques au lieu de quatre.

**Reply by arnolem** [reply]
Ok, je n'avais pas compris cela dans ce sens.
Effectivement, c'est également une remarque que j'ai eu à la lecture. Ca fait super redondant alors que c'est la même pratique.

On peut même aller plus loin en minifiant/compressant le JSON, XML, SVG, CSV, ...
C'est une pratique que je vois très rarement alors que le gain est important.

**Reply by Christophe-Cl** [reply]
Pour moi Minifier et Compresser sont bien 2 choses différentes, aux résultats différents, donc aux termes bien distincts :)
On minifie un fichier CSS/JS/autre mais on compresse également une image, un doc PDF, une vidéo tout comme on compresse les données serveur <-> navigateur. 
Pour ne pas créer de confusion, il me semble opportun de conserver la distinction.

**Reply by rlemaire** [reply]
C'est bien ce que je dis @Christophe-Cl  : «  Du coup cela fait deux pratiques au lieu de quatre. »

**Reply by Christophe-Cl** [reply]
Toutafé ! (je plussoyais) :)

**Reply by Fred-artwai** [reply]
Toutafé d'accord juste 2 BPs une pour la minification et l'autre pour la compression.
A propos de la compression, il n'y a pas que le GZIP , il y aussi le Brotli qui est encore plus léger je pense qu'il est bon de le mentionner.
Attention, toutefois pour des tout petits fichiers la compression est inutiles et certains fichier comme le woff2 sont déjà compressés par nature.

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/140
https://github.com/cnumr/best-practices/issues/141
